web-ng/src/app/components/feature-panel/feature-panel.component.ts

```diff
@@ -79,6 +80,34 @@ export class FeaturePanelComponent implements OnInit, OnDestroy {
             .pipe(map(feature => project.layers.get(feature.layerId)!))
         )
       );
+    this.photoUrls = new Map();
+    this.observations$.forEach((observations) => {
+      observations.forEach(observation => {
+        this.getFields(observation).forEach(
+          field => {
+            if (field.type == FieldType.PHOTO) {
+              if (observation.responses?.get(field.id)?.value as string) {
+                this.fillPhotoURL(field.id, observation.responses?.get(field.id)?.value as string);
+              }
+            }
+          }
+        );
+      });
+    });
+  }
+
+  openUrlInNewTab(url: string) {
+    window.open(url, '_blank');
+  }
+
+  fillPhotoURL(fieldId: string, storageFilePath: string) {
+    this.dataStoreService.getImageDownloadURL(storageFilePath)
+      .then((url) => {
+        this.photoUrls.set(fieldId, url);
```

Comment:
Iiuc `photoUrls` would contain all photo URLs for all fields in all observations for features shown during the lifecycle of this component. Instead, would it be easier, cleaner, and safer to instead add an optional photoUrl field to `Response`?

Reply:
We spoke: Please double check that feature panel is rebuilt each time a new feature is selected

Reply:
so Panel constructor is not called each time but this.observations$.forEach is called each time observations$
<img width="1434" alt="Screen Shot 2021-05-06 at 2 59 46 PM" src="https://user-images.githubusercontent.com/222107/117352606-2e8bf000-ae7d-11eb-81d2-afe1db6c115c.png">
 change